contributors/design-proposals/api-machinery/customresource-conversion-webhook.md

```diff
@@ -178,20 +178,27 @@ type ConversionRequest struct {
   UID types.UID
   // The version to convert given object to. E.g. "stable.example.com/v1"
   APIVersion string
-  // Object is the CRD object to be converted.
+  // Object is the CRD object or list of objects to be converted.
   Object runtime.RawExtension
+  // IsList indicates that this is a List operation and Object contains a list of items.
+  IsList bool
 }
 
 type ConversionResponse struct {
   // UID is an identifier for the individual request/response.
   // This should be copied over from the corresponding ConversionRequest.
   UID types.UID
-  // ConvertedObject is the converted version of request.Object.
-  ConvertedObject runtime.RawExtension
+  // ConvertedObject is the converted version of request.Object if the Result field is nil.
+  // +optional
+  ConvertedObject *runtime.RawExtension
+  // Result contains extra details into why a conversion request was failed. If it is not nil, it means
+  // the conversion has been failed.
+  // +optional
+  Result *metav1.Status
 }
 ```
 
-If the conversion is failed, the webhook should fail the HTTP request with a proper error code and message that will be used to create a status error for the original API caller.
+If the conversion is failed, the webhook is prefered to fill up the Result field of the response with a proper error code and message instead of returning failure HTTP codes.
```

Comment:
Perhaps: "Instead of returning failure HTTP status codes, the preferred way for a webhook to indicate a conversion failure is to fill the `result` field of the response with an error code and message. Non-200 status return codes will be taken to mean there was a problem with the webhook rather than a problem with the object to be converted."

Reply:
Fixed that in the implementation and will fix it here too.